Makefile

```diff
@@ -1,38 +1,52 @@
+AR=ar
 CC=clang
-CXX=clang
-LD=ld.lld
-RUST=rustc
+CXX=clang++
+
+ifeq ($(shell uname),Darwin)
+    LDFLAGS := -Wl,-dead_strip
+else
+    LDFLAGS := -Wl,--gc-sections -lpthread -ldl
+endif
+
+ZIGFLAGS:=-fPIC --bundle-compiler-rt
+
+all: target/link-all-languages
+	@printf "\x1b[0;36mRunning ...\e[0m\n"
+	@target/link-all-languages
+	@printf "\e[92mDone\e[0m\n"
+
+target:
+	@mkdir -p $@
+
+target/link-all-languages: target/main.o target/debug/libhello_rust.a target/debug/libhello_cpp.a target/debug/libhello_c.a target/debug/libhello_zig.a
+	@$(CC) -o $@ $^ $(LDFLAGS) -lstdc++
+
+target/debug/libhello_rust.a: src/lib.rs Cargo.toml
+	@printf "\x1b[0;36mBuilding $@ ...\e[0m\n"
+	@cargo build
+	@printf "\e[92mBuilt $@\e[0m\n"
+
+target/debug/libhello_cpp.a: src/lib.cpp
+	@printf "\x1b[0;36mBuilding $@ ...\e[0m\n"
+	@$(CXX) -c $^ -o target/debug/libhello_cpp.o
+	@$(AR) rcs $@ target/debug/libhello_cpp.o
+	@printf "\e[92mBuilt $@\e[0m\n"
+
+target/debug/libhello_c.a: src/lib.c
+	@printf "\x1b[0;36mBuilding $@ ...\e[0m\n"
+	@$(CC) -c $^ -o target/debug/libhello_c.o
+	@$(AR) rcs $@ target/debug/libhello_c.o
+	@printf "\e[92mBuilt $@\e[0m\n"
+
+target/debug/libhello_zig.a: src/lib.zig
+	@printf "\x1b[0;36mBuilding $@ ...\e[0m\n"
+	@zig build-lib $^ --output-dir target/debug --name hello_zig $(ZIGFLAGS)
+	@printf "\e[92mBuilt $@\e[0m\n"
+
+target/main.o: src/main.c | target
+	@printf "\x1b[0;36mBuilding $@ ...\e[0m\n"
+	@$(CC) -o $@ -c $<
+	@printf "\e[92mBuilt $@\e[0m\n"
```

Comment:
I don't like hiding the commands in favor of a bunch of printf statements

Reply:
Should I just show or show + rm printfs?